docs/Building_Diagrams/data_stores.md

```diff
@@ -10,3 +10,141 @@ It can represent a database, file system, or any other storage mechanism.
 You might use a Data Store when it is not sufficient for data to be accessible in just a single process instance, but it needs to be shared across process instances.
 If you a have a use case where you need to store data and access it from multiple different process instances, you could also consider using a Service Task to contact a database external to SpiffWorkflow, either via a database library in a connector or using a database via API.
 All of these mechanisms work well in SpiffWorkflow, so the choice will depend on your storage and performance requirements.
+
+## Types of Data Store
+### KKV Data Store in BPMN
+
+Key-Key-Value (KKV) Data Stores extend the traditional key-value model by introducing an additional key layer. This structure enables more complex and hierarchical data storage solutions, making them particularly suited for BPMN (Business Process Model and Notation) processes that manage multifaceted data relationships.
+
+- **Structure**: A KKV data store organizes data into two levels of keys before reaching the actual value, e.g., `Genre -> Movie -> Attributes`. This structure is particularly beneficial for categorizing and accessing nested data efficiently.
+
+- **Use Cases**: Ideal for scenarios requiring rich, structured data access within workflows, such as managing inventories, categorizing user profiles, or handling complex hierarchical data like the provided movies example.
+
+#### BPMN Integration
+
+Integrating KKV data stores within BPMN diagrams involves modeling tasks that interact with the data store for CRUD (Create, Read, Update, Delete) operations. 
+
+Here's how to depict such interactions using a BPMN example focused on movie data management:
+
+### BPMN Example: Movie Data Management
+
+#### Process Overview
+![KKV data_store](images/DataStore_KKV.png)
+
+1. **Start Event**: Initiates the movie data management workflow.
+
+2. **Script Task: Populate Data Store**
+    - Uses a script to populate the "Movies" data store with detailed information about characters in various movies, demonstrating the KKV structure (Director -> Movie -> Character Attributes).
+
+3. **Script Task: Extract Single Record**
+    - Retrieves specific movie details from the "Movies" data store.
+    - A script extracts information about "Django Unchained" using the KKV path, illustrating how to access nested data.
+    - **Script**: `django = movies["Quentin Tarantino"]["Django Unchained"]`
```

Comment:
The kkv data store places a function pointer in task data when a read arrow is drawn from the data store. This allows just reading data for a top level/secondary key combination without loading all the data. In this case the read would look like `django = movies("Quentin Tarantino", "Django Unchained")`.